crates/sui-framework/packages/sui-framework/sources/coin.move

```diff
@@ -323,6 +370,79 @@ module sui::coin {
         include Burn<T>;
     }
 
+    /// Freezes the given address, preventing it
+    /// from interacting with the coin as an input to a transaction.
+    public fun freeze_address<T>(
```

Comment:
I think these functions all need to go into freezer.move, with an argument for type index. Otherwise we would end up duplicating the code for future types

Reply:
I can do this but it gets potentially a bit strange because not all instances will have a type `T` necessarily?

Reply:
The FreezeCap is just for Coin. I had imagined that each index's type would manage it's cap. 
So basically one shared raw implementation that each type (coin or token) builds on top of, and manages its own access control.